src/hosting/normalizedHostingConfigs.ts

```diff
@@ -24,16 +24,35 @@ function filterOnly(configs: HostingConfig[], onlyString: string): HostingConfig
     .filter((target) => target.startsWith("hosting:"))
     .map((target) => target.replace("hosting:", ""));
 
-  // Check to see that all the hosting deploy targets exist in the hosting config
-  onlyTargets.forEach((onlyTarget) => {
-    if (!configs.some((config) => config.target === onlyTarget)) {
-      throw new FirebaseError(`Hosting target ${bold(onlyTarget)} not detected in firebase.json`);
+  const configsBySite = new Map<string, HostingConfig>();
+  const configsByTarget = new Map<string, HostingConfig>();
+  for (const c of configs) {
+    if (c.site) {
+      configsBySite.set(c.site, c);
     }
-  });
+    if (c.target) {
+      configsByTarget.set(c.target, c);
+    }
+  }
+
+  const filteredConfigs: HostingConfig[] = [];
+  // Check to see that all the hosting deploy targets exist in the hosting
+  // config as either `site`s or `target`s.
+  for (const onlyTarget of onlyTargets) {
+    if (configsBySite.has(onlyTarget)) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Why is this needed?

Reply:
Oh! Yeah, so Typescript doesn't know that if you do `if (Map.has(thing)) Map.get(thing)`, the `get` is guaranteed to exist (and have the type as defined by the Map): https://github.com/microsoft/TypeScript/issues/9619